config/base/derive/src/configurable.rs

```diff
@@ -0,0 +1,228 @@
+use proc_macro::TokenStream;
+use quote::{format_ident, quote};
+use syn::{Attribute, Ident, Type, TypePath};
+
+use super::utils::{
+    extract_field_attrs, extract_field_idents, extract_field_types, field_has_as_str_attr,
+    field_has_inner_attr, field_has_option_type, gen_field_env, gen_lvalues, get_env_prefix,
+    get_inner_type, StructWithFields,
+};
+
+pub(super) fn impl_configurable(ast: &StructWithFields) -> TokenStream {
+    let StructWithFields {
+        ident: parent_name,
+        vis,
+        generics,
+        ..
+    } = ast;
+    let proxy_name = format_ident!("{}Proxy", parent_name);
```

Comment:
```suggestion
    let proxy_name = format_ident!("_Proxy_{}", parent_name);
```
This conveys that the type isn't supposed to be usable by the programmer.

Reply:
But wouldn't the `_Proxy` types be the entrypoint? I saw it as in `cli/src/lib.rs` we would be calling `ConfigurationProxy::from_path()` instead of the current one.

Reply:
My perception was that creating the config was a separate procedure and that the proxy structures were going to be local to the `Configuration` logic. 

Using them directly inside `iroha_cli` is acceptable for now, but I'd much rather the logic were internal to the `iroha_config` crate.

Reply:
So e.g. when we decorate a `Configuration` struct with `#[derive(Combine)]`, it implements methods such as `load_environment()` and `from_path()` on the parent struct, but internally creates a `_Proxy` and calls those methods on the latter?